File: Java/Data Structure/Sorting/program2.java
//Selection Sort

import java.util.*;

class ArrayX
{
    public int Arr[];

    public ArrayX(int iSize)
    {
        Arr = new int[iSize];
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter elements");
        for(int i= 0;i <Arr.length; i++)
        {
            Arr[i] = sobj.nextInt();
        }
    }

    public void Display()
   {
        System.out.println("Elements of the array are");
        for(int i= 0;i <Arr.length; i++)
        {
           System.out.print(Arr[i]+"  ");
        }
        System.out.println();
   }

   public void SelectionSort()
   {
        int i = 0, j = 0, min_index = 0, temp = 0;

        for(i = 0; i < Arr.length-1; i++)
        {
            min_index = i;
            for(j = i+1; j < Arr.length; j++)
            {
                if(Arr[min_index] > Arr[j])
                {
                    min_index = j;
                }
            }
            if(i != min_index)
            {
                temp = Arr[i];
                Arr[i] = Arr[min_index];
                Arr[min_index] = temp;
            }

        }
   }
}

class program2
{
    public static void main(String arr[])
    {
            Scanner sobj = new Scanner(System.in);

            System.out.println("Enter the size of array : ");
            int size = sobj.nextInt();

            ArrayX aobj = new ArrayX(size);
            aobj.Accept();
            aobj.Display();

            aobj.SelectionSort();
            System.out.println("Data after sorting");
            aobj.Display();
    }
}

File: C++/Data Structure/Sorting/program2.cpp
//Selection Sort

#include<iostream>
using namespace std;

class ArrayX
{
    public:
        int *Arr;
        int iSize;

        ArrayX(int value)
        {
            iSize = value;
            Arr = new int[iSize];
        }

        ~ArrayX()
        {
            delete [] Arr;
        }

        void Accept()
        {
            cout<<"Enter the values : "<<endl;
            for(int i = 0; i < iSize; i++)
            {
                cin>>Arr[i];
            }
        }

        void Display()
        {
            cout<<"Elements of the array are: "<<endl;

            for(int i = 0; i < iSize; i++)
            {
                cout<<Arr[i]<<"\t";
            }
            cout<<endl;
        }

        void SelectionSort()
        {
            int i = 0, j = 0, min_index = 0, temp = 0;

            for(i = 0; i < iSize-1; i++)
            {
                min_index = i;
                for(j = i+1; j < iSize; j++)
                {
                    if(Arr[min_index] > Arr[j])
                    {
                        min_index = j;
                    }
                }
                if(i != min_index)
                {
                    temp = Arr[i];
                    Arr[i] = Arr[min_index];
                    Arr[min_index] = temp;
                }

            }
        }
};

int main()
{
    int iLength = 0;

    cout<<"Enter the size of array"<<endl;
    cin>>iLength;

    ArrayX obj1(iLength);

    obj1.Accept();
    obj1.Display();

    obj1.SelectionSort();
    cout<<"Data after sorting"<<endl;
    obj1.Display();

    return 0;
}